### PR TITLE
Various signer tidy-ups in preparation

### PR DESCRIPTION
**Pull request by daviddrysdale**
Probably best reviewed commit-by-commit.

Helps with issue #419

### CODE REVIEW

File: server/log_operation_manager.go [hunk 1]
```diff
@@ -70,41 +71,42 @@ func NewLogOperationManager(registry extension.Registry, batchSize, numWorkers i
 	}
 }
 
-func (l LogOperationManager) getLogsAndExecutePass(ctx context.Context) bool {
+// getLogs returns the current set of active log IDs.
```
Comment (Martin2112) [inline server/log_operation_manager.go]: getLogIDs returns ...

File: server/log_operation_manager.go [hunk 2]
```diff
@@ -115,14 +117,18 @@ func (l LogOperationManager) OperationLoop(ctx context.Context) {
 	// Outer loop, runs until terminated
 	for {
 		// TODO(alcutter): want a child context with deadline here?
```
Comment (Martin2112) [inline server/log_operation_manager.go]: This todo may be out of date?

File: server/log_operation_manager.go [hunk 3]
```diff
@@ -96,8 +98,65 @@ func (l LogOperationManager) getLogsAndExecutePass(ctx context.Context) error {
...
+			}
+		}()
+	}
+
+	// Wait for the workers to consume all of the logIDs
+	wg.Wait()
+	d := time.Now().Sub(startBatch).Seconds()
+
+	mu.Lock()
+	glog.V(1).Infof("Group run completed in %.2f seconds: %v succeeded, %v failed, %v items processed", d, successCount, len(logIDs)-successCount, itemCount)
```
Comment (Martin2112) [inline server/log_operation_manager.go]: Not sure of the reason for the lock as the wait group is done. Keeping lock analysis tools happy?

File: server/log_operation_manager.go [hunk 4]
```diff
@@ -96,8 +98,65 @@ func (l LogOperationManager) getLogsAndExecutePass(ctx context.Context) error {
...
+				ctx = util.NewLogContext(ctx, logID)
+				start := time.Now()
+				count, err := l.logOperation.ExecutePass(ctx, logID, &l.info)
+
+				if err == nil {
+					if count > 0 {
+						d := time.Now().Sub(start).Seconds()
+						glog.Infof("%v: processed %d items in %.2f seconds (%.2f qps)", logID, count, d, float64(count)/d)
+					} else {
+						glog.V(1).Infof("%v: no items to process", logID)
```
Comment (Martin2112) [inline server/log_operation_manager.go]: I think we might have removed this logging when no work is done to cut down noise on long test runs. OTOH this is V(1) and the one we deleted wasn't.

File: server/log_operation_manager.go (hunk 1 above)
Comment (daviddrysdale) [inline server/log_operation_manager.go]: Done.

File: server/log_operation_manager.go (hunk 2 above)
Comment (daviddrysdale) [inline server/log_operation_manager.go]: Not sure -- this context doesn't have a deadline so I'm guessing not.

File: server/log_operation_manager.go (hunk 4 above)
Comment (daviddrysdale) [inline server/log_operation_manager.go]: These logs were moved across from the `SequencerManager`, so maybe the deletion was actually a conversion to V(1)?

Comment (Martin2112) [inline server/log_operation_manager.go]: That's possible. I think V(1) is fine.

File: server/log_operation_manager.go (hunk 3 above)
Comment (daviddrysdale) [inline server/log_operation_manager.go]: This was copied over from the `SequencerManager` code, but you're right -- it shouldn't be necessary now.

**Comment by daviddrysdale** [comment]
PTAL -- I've added 6 more commits to try to understand/fix the flaky client tests.

File: client/client_test.go [hunk 5]
```diff
@@ -160,14 +154,6 @@ func TestAddLeaf(t *testing.T) {
...
 	} {
 		client := New(logID, test.client, testonly.Hasher, env.PublicKey)
-		{
-			ctx, cancel := context.WithDeadline(ctx, time.Now().Add(timeout))
-			defer cancel()
-			if err, want := client.AddLeaf(ctx, []byte(test.desc)), codes.DeadlineExceeded; grpc.Code(err) != want {
-				t.Errorf("AddLeaf(%v): %v, want, %v", test.desc, err, want)
-				continue
-			}
-		}
```
Comment (daviddrysdale) [inline client/client_test.go]: I'm a bit uncertain about this change as I don't understand what the point of the code I removed is (Chesterton's Fence)!  @gdbelvin, what was this supposed to do?

Comment (Martin2112) [inline client/client_test.go]: Bonus points for use of Chesterton's Fence.

File: server/validate.go [hunk 6]
```diff
@@ -23,56 +23,56 @@ import (
...
-		return grpc.Errorf(codes.InvalidArgument, "LeafIndex: %v >= TreeSize: %v, want < ", req.LeafIndex, req.TreeSize)
+		return grpc.Errorf(codes.InvalidArgument, "GetInclusionProofRequest.LeafIndex: %v >= TreeSize: %v, want < ", req.LeafIndex, req.TreeSize)
 	}
 	return nil
 }
 
 func validateGetInclusionProofByHashRequest(req *trillian.GetInclusionProofByHashRequest) error {
 	if req.TreeSize <= 0 {
-		return grpc.Errorf(codes.InvalidArgument, "TreeSize: %v, want > 0", req.TreeSize)
+		return grpc.Errorf(codes.InvalidArgument, "GetInclusionProofBuHashRequest.TreeSize: %v, want > 0", req.TreeSize)
```
Comment (Martin2112) [inline server/validate.go]: Typo.

Comment (daviddrysdale) [inline server/validate.go]: Well spotted.

**Pull request merged**